### PR TITLE
[zh-cn]: Update translation of number.overview

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#15736

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 1]
```diff
@@ -5,26 +5,29 @@ slug: Web/JavaScript/Reference/Global_Objects/Number
...
+255; // two-hundred and fifty-five
+255.0; // same number
+255 === 255.0; // true
+255 === 0xff; // true (hexadecimal notation)
+255 === 0b11111111; // true (binary notation)
+255 === 0.255e3; // true (decimal exponential notation)
 ```
 
-在 JavaScript 代码中，像 `37` 这样的数值字面值是一个浮点值，而不是整数。在常见的日常使用中，没有单独的整数类型。（JavaScript 也有 {{jsxref("BigInt")}} 类型，但它不是为取代日常使用的 Number 而设计的。`37` 仍然是一个数值，而不是 BigInt。）
+在 JavaScript 代码中，像 `37` 这样的数字字面量是浮点数值，而不是整数。在常见的日常使用中，JavaScript 没有单独的整数类型。（JavaScript还有一个 {{jsxref("BigInt")}} 类型，但它并不是为了取代 Number 而设计的，`37` 仍然是一个数字，而不是一个 BigInt。）
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:25]:
```suggestion
在 JavaScript 代码中，像 `37` 这样的数字字面量是浮点数值，而不是整数。在常见的日常使用中，JavaScript 没有单独的整数类型。（JavaScript 还有一个 {{jsxref("BigInt")}} 类型，但它并不是为了取代 Number 而设计的，`37` 仍然是一个数字，而不是一个 BigInt。）
```

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 2]
```diff
@@ -5,26 +5,29 @@ slug: Web/JavaScript/Reference/Global_Objects/Number
...
 ```js
-123; // one-hundred twenty-three
-123.0; // same
-123 === 123.0; // true
+255; // two-hundred and fifty-five
+255.0; // same number
+255 === 255.0; // true
+255 === 0xff; // true (hexadecimal notation)
+255 === 0b11111111; // true (binary notation)
+255 === 0.255e3; // true (decimal exponential notation)
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:22]:
```suggestion
255; // 二百五十五
255.0; // 相同的数字
255 === 255.0; // true
255 === 0xff; // true（十六进制表示）
255 === 0b11111111; // true（二进制表示）
255 === 0.255e3; // true（十进制指数记数法）
```

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 3]
```diff
@@ -33,49 +36,49 @@ Number(undefined); // NaN
...
 
-JavaScript 的 `Number` 类型是一个[双精度 64 位二进制格式 IEEE 754](https://zh.wikipedia.org/wiki/雙精度浮點數) 值，类似于 Java 或者 C# 中的 `double`。这意味着它可以表示小数值，但是存储数值的大小和精度有一些限制。简而言之，IEEE 754 双精度浮点数使用 64 位来表示 3 个部分：
+JavaScript 的 `Number` 类型是一个[双精度 64 位二进制格式 IEEE 754](https://zh.wikipedia.org/wiki/雙精度浮點數) 值，类似于 Java 或者 C# 中的 `double`。这意味着它可以表示小数值，但是存储的数字的大小和精度有一些限制。简而言之，IEEE 754 双精度浮点数使用 64 位来表示 3 个部分：
 
-- 1 比特*符号位*（sign）（正数或者负数）
-- 11 比特*指数位*（exponent）（-1022 到 1023）
-- 52 比特*尾数位*（mantissa）（表示 0 和 1 之间的数值）
+- 1 位用于表示*符号*（正数或者负数）
+- 11 位用于表示*指数*（-1022 到 1023）
+- 52 位用于表示*尾数*（表示 0 和 1 之间的数值）
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:43]: cloud we reserve the English annotates: sign, exponect and mantissa, as we have a mathematical formula below

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 4]
```diff
@@ -33,49 +36,49 @@ Number(undefined); // NaN
...
 
-尾数（也称为*有效位数*）是表示实际值的数值（有效数值）部分。指数是尾数应乘以的 2 的幂。将其视为科学计数法：
+尾数（也称为*有效数*）是表示实际值（有效数字）的数值部分。指数是尾数应乘以的 2 的幂次。将其视为科学计数法：
 
 <math display="block"><semantics><mrow><mtext>Number</mtext><mo>=</mo><mo stretchy="false">(</mo><mrow><mo>−</mo><mn>1</mn></mrow><msup><mo stretchy="false">)</mo><mtext>sign</mtext></msup><mo>⋅</mo><mo stretchy="false">(</mo><mn>1</mn><mo>+</mo><mtext>mantissa</mtext><mo stretchy="false">)</mo><mo>⋅</mo><msup><mn>2</mn><mtext>exponent</mtext></msup></mrow><annotation encoding="TeX">\text{Number} = ({-1})^{\text{sign}} \cdot (1 + \text{mantissa}) \cdot 2^{\text{exponent}}</annotation></semantics></math>
 
-尾数使用 52 比特存储，在二进制小数中解释为 `1.…` 之后的数值。因此，尾数的精度是 2<sup>-52</sup>（可以通过 {{jsxref("Number.EPSILON")}} 获得），或者十进制小数点后大约 15 到 17 位；高于该精度级别的算术需要[舍入](https://en.wikipedia.org/wiki/Floating-point_arithmetic#Representable_numbers,_conversion_and_rounding)。
+尾数使用 52 比特存储，在二进制小数中解释为 `1.…` 之后的数字。因此，尾数的精度是 2<sup>-52</sup>（可以通过 {{jsxref("Number.EPSILON")}} 获得），或者十进制数小数点后大约 15 到 17 位；超过这个精度的算术会收到[舍入](https://zh.wikipedia.org/wiki/%E6%B5%AE%E7%82%B9%E6%95%B0)的影响。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:49]: Lets use "准确性" section instead

```suggestion
尾数使用 52 比特存储，在二进制小数中解释为 `1.…` 之后的数字。因此，尾数的精度是 2<sup>-52</sup>（可以通过 {{jsxref("Number.EPSILON")}} 获得），或者十进制数小数点后大约 15 到 17 位；超过这个精度的算术会受到[舍入](https://zh.wikipedia.org/wiki/浮点数#准确性)的影响。
```

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 5]
```diff
@@ -33,49 +36,49 @@ Number(undefined); // NaN
...
 
-一个数值可以容纳的最大值是 2<sup>1024</sup> - 1（基于二进制的指数为 1023，尾数为 0.1111...），这可以通过 {{jsxref("Number.MAX_VALUE")}} 获得。大于该值的数值，被特殊数常数 {{jsxref("Infinity")}} 取代。
+一个数值可以容纳的最大值是 2<sup>1024</sup> - 1（基于二进制，指数为 1023，尾数为 0.1111…），可以通过 {{jsxref("Number.MAX_VALUE")}} 获得。超过这个值的数会被替换为特殊的数值常量{{jsxref("Infinity")}}。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:51]:
```suggestion
一个数值可以容纳的最大值是 2<sup>1024</sup> - 1（指数为 1023，尾数为基于二进制的 0.1111…），可以通过 {{jsxref("Number.MAX_VALUE")}} 获得。超过这个值的数会被替换为特殊的数值常量 {{jsxref("Infinity")}}。
```

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 6]
```diff
@@ -33,49 +36,49 @@ Number(undefined); // NaN
...
   - `Infinity` 和 `-Infinity` 被当作是字面量。在实际代码中，它们是全局变量。
-  - 空字符串或仅空格字符串转换为 `0`。
-  - 不允许使用[数字分隔符](/zh-CN/docs/Web/JavaScript/Reference/Lexical_grammar#numeric_separators)。
-- [BigInt](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/BigInt) 抛出 {{jsxref("TypeError")}}，以防止意外的强制隐式转换损失精度。
+  - 空字符串或仅包含空格的字符串转换为 `0`。
+  - 不允许使用[数字分隔符](/zh-CN/docs/Web/JavaScript/Reference/Lexical_grammar#数值分隔符)。
+- [BigInt](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/BigInt) 抛出 {{jsxref("TypeError")}}，以防止意外的强制隐式转换导致精度损失。
 - [Symbol](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Symbol) 抛出 {{jsxref("TypeError")}}。
-- 对象首先按顺序调用 [`[@@toPrimitive]()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive)（将 `"number"` 作为 hint）、`valueOf()` 和 `toString()` 方法将其[转换为原始值](/zh-CN/docs/Web/JavaScript/Data_structures#强制原始值转换)。然后将生成的原始值转换为数值。
+- 对象首先通过按顺序调用它们的 [`[@@toPrimitive]()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive)（带有 `"number"` 作为提示）、`valueOf()` 和 `toString()` 方法将其[转换为原始值](/zh-CN/docs/Web/JavaScript/Data_structures#强制原始值转换)。然后将得到的原始值转换为数字。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:74]:
```suggestion
- 对象首先通过按顺序调用它们的 [`[@@toPrimitive]()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive)（使用 `"number"` 提示）、`valueOf()` 和 `toString()` 方法将其[转换为原始值](/zh-CN/docs/Web/JavaScript/Data_structures#强制原始值转换)。然后将得到的原始值转换为数字。
```

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 7]
```diff
@@ -33,49 +36,49 @@ Number(undefined); // NaN
...
 有两种方法可以在 JavaScript 中实现几乎相同的效果。
 
 - [一元加](/zh-CN/docs/Web/JavaScript/Reference/Operators/Unary_plus)：`+x` 完全按照上面的数值强制转换步骤来转换 `x`。
-- [`Number()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Number/Number) 函数：`Number(x)` 使用相同的算法转换 `x`，除了 [BigInt](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/BigInt) 不会抛出 {{jsxref("TypeError")}}，而是返回它的 Number 值，并且可能损失精度。
+- [`Number()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Number/Number) 函数：`Number(x)` 使用相同的算法转换 `x`，除了 [BigInt](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/BigInt) 不会抛出 {{jsxref("TypeError")}}，而是返回其数字值，并且可能导致精度损失。
 
-{{jsxref("Number.parseFloat()")}} 和 {{jsxref("Number.parseInt()")}} 与 `Number()` 相似，但只转换字符串，并且解析规则略有不同。例如，`parseInt()` 无法识别小数点，`parseFloat()` 无法识别 `0x` 前缀。
+{{jsxref("Number.parseFloat()")}} 和 {{jsxref("Number.parseInt()")}} 与 `Number()` 相似，但只能转换字符串，并且解析规则略有不同。例如，`parseInt()` 无法识别小数点，`parseFloat()` 无法识别 `0x` 前缀。
 
 #### 整数转换
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:84]: comment for the next line: should we use "通过丢弃小数部分"?

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 8]
```diff
@@ -85,23 +88,31 @@ JavaScript 的 `Number` 类型是一个[双精度 64 位二进制格式 IEEE 754
 
 #### 固定宽度数值转换
 
-JavaScript 有一些较低级别的函数，用于处理整数的二进制编码，最值得注意的是[按位运算](/zh-CN/docs/Web/JavaScript/Reference/Operators#bitwise_shift_operators)和 {{jsxref("TypedArray")}} 对象。按位运算总是将操作数转换为 32 位整数。在这些情况下，将值转换为数值后，然后首先[截断](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Math/trunc)小数部分，然后在整数的二进制的补码编码中取最低位，将数值归一化为给定的宽度。
+JavaScript 有一些较低级别的函数，用于处理整数的二进制编码，最值得注意的是[按位运算](/zh-CN/docs/Web/JavaScript/Reference/Operators#位移运算符)和 {{jsxref("TypedArray")}} 对象。按位运算总是将操作数转换为 32 位整数。在这些情况下，将值转换为数字后，数字将首先[截断](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Math/trunc)小数部分，然后再取整数的二进制的补码编码中的最低位来将数值归一化为给定的宽度。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:91]: "lowest bits" isn't only one bit.

```suggestion
JavaScript 有一些较低级别的函数，用于处理整数的二进制编码，最值得注意的是[按位运算](/zh-CN/docs/Web/JavaScript/Reference/Operators#位移运算符)和 {{jsxref("TypedArray")}} 对象。按位运算总是将操作数转换为 32 位整数。在这些情况下，将值转换为数字后，数字将首先[截断](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Math/trunc)小数部分，然后再取整数的二进制的补码编码中的最低几位来将数值标准化为给定的宽度。
```

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 9]
```diff
@@ -85,23 +88,31 @@ JavaScript 的 `Number` 类型是一个[双精度 64 位二进制格式 IEEE 754
...
 new Int32Array([1.1, 1.9, -1.1, -1.9]); // Int32Array(4) [ 1, 1, -1, -1 ]
-new Int8Array([257, -257]); // Int8Array(1) [ 1, -1 ]
-// 257 = 0001 0000 0001 = 0000 0001 (mod 2^8) = 1
-// -257 = 1110 1111 1111 = 1111 1111 (mod 2^8) = -1 (as signed integer)
-new Uint8Array([257, -257]); // Uint8Array(1) [ 1, 255 ]
-// -257 = 1110 1111 1111 = 1111 1111 (mod 2^8) = 255 (as unsigned integer)
+
+new Int8Array([257, -257]); // Int8Array(2) [ 1, -1 ]
+// 257 = 0001 0000 0001
+//     =      0000 0001 (mod 2^8)
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:98]:
```suggestion
//     =      0000 0001（模 2^8）
```

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 10]
```diff
@@ -85,23 +88,31 @@ JavaScript 的 `Number` 类型是一个[双精度 64 位二进制格式 IEEE 754
...
+//     = 1
+// -257 = 1110 1111 1111
+//      =      1111 1111 (mod 2^8)
+//      = -1 (as signed integer)
+
+new Uint8Array([257, -257]); // Uint8Array(2) [ 1, 255 ]
+// -257 = 1110 1111 1111
+//      =      1111 1111 (mod 2^8)
+//      = 255 (as unsigned integer)
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:107]:
```suggestion
//      =      1111 1111（模 2^8）
//      = 255（无符号整数）
```

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 11]
```diff
@@ -123,23 +134,29 @@ new Uint8Array([257, -257]); // Uint8Array(1) [ 1, 255 ]
   - : 特殊的负无穷大值，在溢出时返回该值。
 - {{jsxref("Number.POSITIVE_INFINITY")}}
   - : 特殊的正无穷大值，在溢出时返回该值。
-- {{jsxref("Number.prototype")}}
-  - : Number 对象上允许的额外属性。
+  -
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:137]:
```suggestion
```

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 12]
```diff
@@ -148,13 +165,13 @@ new Uint8Array([257, -257]); // Uint8Array(1) [ 1, 255 ]
...
 - {{jsxref("Number.prototype.toPrecision()")}}
   - : 返回数值使用定点表示法或指数表示法至指定精度的字符串。
 - {{jsxref("Number.prototype.toString()")}}
-  - : 返回一个代表给定对象的字符串，基于指定的*基数*（radix）。覆盖 {{jsxref("Object.prototype.toString()")}} 方法。
+  - : 返回一个代表给定对象的字符串，基于指定的*基数*（radix）。重写了 {{jsxref("Object.prototype.toString()")}} 方法。
 - {{jsxref("Number.prototype.valueOf()")}}
-  - : 返回指定对象的原始值。覆盖 {{jsxref("Object.prototype.valueOf()")}} 方法。
+  - : 返回指定对象的原始值。重写了 {{jsxref("Object.prototype.valueOf()")}} 方法。
 
 ## 示例
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:177]: Comment for line 199, could you remove the second "时"

File: files/zh-cn/web/javascript/reference/global_objects/number/index.md [hunk 13]
```diff
@@ -226,6 +243,6 @@ Number("-Infinity"); // -Infinity
 
 - [在 `core-js` 中现代 `Number` 行为的 Polyfill（支持二进制和八进制字面量）](https://github.com/zloirock/core-js#ecmascript-number)
 - {{jsxref("NaN")}}
-- [算术操作符](/zh-CN/docs/Web/JavaScript/Reference/Operators#arithmetic_operators)
-- {{jsxref("Math")}} 全局对象
-- 任意精度的整数：{{jsxref("BigInt")}}
+- [算术操作符](/zh-CN/docs/Web/JavaScript/Reference/Operators#算术运算符)
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/number/index.md:246]:
```suggestion
- [算术运算符](/zh-CN/docs/Web/JavaScript/Reference/Operators#算术运算符)
```

**Review by yin1999 — APPROVED** [review]
Thank you @JasonLamv-t :+1:

**Pull request merged**